File: 프로그래머스/3/389481. 봉인된 주문/봉인된 주문.java
import java.util.*;
class Solution {
    public String solution(long n, String[] bans) {
        Arrays.sort(bans, new Comparator<String>(){
            @Override
            public int compare(String s1, String s2){
                if(s1.length() == s2.length()){
                    return s1.compareTo(s2);
                }
                return s1.length() - s2.length();
            }
        });
        
        for(String s : bans){
            long numsOrder = getOrder(s);
            if(numsOrder <= n){
                n ++;
            }
        }
        
        return getAnswer(n);
    }
    
     public String getAnswer(long n) {
        StringBuilder sb = new StringBuilder();
        while (n > 0) {
            long remainder = n % 26;
            if (remainder == 0) {
                sb.append('z');
                n = (n / 26) - 1;
            } else { 
                sb.append((char)('a' + remainder - 1));
                n = n / 26;
            }
        }
        return sb.reverse().toString();
    }
    
    public long getOrder(String s){
        long tmp = 0;
        for(int i = 0; i < s.length(); i ++){
            tmp += (s.charAt(i) - 'a' + 1) * Math.pow(26, s.length() - i - 1);
        }
        return tmp;
    }
}

File: 프로그래머스/3/389481. 봉인된 주문/봉인된 주문.py
def solution(n, bans):
    def str_to_num(value):
        cur = 0
        for i in range(len(value)):
            cur += (ord(value[i]) - ord('a') + 1) * (26 ** (len(value) - i - 1))
        return cur - 1
    
    def num_to_str(n):
        tmp = []
        while n >= 0:
            ch = chr(ord('a') + int(n % 26))
            n = (n // 26) - 1
            tmp.append(ch)
            
        tmp.reverse()
        return "".join(map(str, tmp))

    bans.sort(key=lambda x: (len(x), x))
    for b in bans:
        idx = str_to_num(b)
        if idx <= (n-1):
            n += 1
            
    return num_to_str((n-1))